File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityTeleportPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record EntityTeleportPacket(int entityId, int x, int y, int z, byte yaw, byte pitch) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
        os.writeByte(yaw);
        os.writeByte(pitch);
    }

    @Override
    public byte getId() {
        return 0x18;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityTeleportPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityTeleportPacket(val entityId: Int, val x: Int, val y: Int, val z: Int, val yaw: Byte, val pitch: Byte) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId, x, y, z)
        os.writeByte(yaw, pitch)
    }

    override val id: Byte
        get() = 0x18
}
